kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/PaginatedScanFilesIteratorImpl.java

```diff
@@ -15,52 +15,142 @@
  */
 package io.delta.kernel.internal.replay;
 
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+
+import io.delta.kernel.Meta;
 import io.delta.kernel.PaginatedScanFilesIterator;
 import io.delta.kernel.data.FilteredColumnarBatch;
 import io.delta.kernel.data.Row;
+import io.delta.kernel.internal.util.Utils;
 import io.delta.kernel.utils.CloseableIterator;
 import java.io.IOException;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** Implementation of {@link PaginatedScanFilesIterator} */
 public class PaginatedScanFilesIteratorImpl implements PaginatedScanFilesIterator {
 
-  private final CloseableIterator<FilteredColumnarBatch> filteredScanFilesIter;
+  private static final Logger logger =
+      LoggerFactory.getLogger(PaginatedScanFilesIteratorImpl.class);
+
+  /**
+   * Filtered ScanFiles iterator retrieved from baseScan but without batches from skipped log files
```

Comment:
`but without batches from skipped log files` --> worth adding a simple example to really drive this point home! 

`For example, if the previous page fully comsumed sidecar files A and B, and half consumed sidecar C, we expect this iterator to exclude sidecar files A and B`